src/localize.ts

```diff
@@ -60,10 +60,18 @@ export default function localize(
     langStored = localStorage.getItem('selectedLanguage');
   }
 
-  const lang = (langStored || navigator.language.split('-')[0] || DEFAULT_LANG)
+  let lang = (langStored || navigator.language.split('-')[0] || DEFAULT_LANG)
     .replace(/['"]+/g, '')
     .replace('-', '_');
 
+  if (lang === 'zh_Hans') {
+    lang = 'zh_CN';
+  }
+
+  if (lang === 'zh_Hant') {
+    lang = 'zh_TW';
+  }
+
```

Comment:
Can you explain this logic? Why is neccessary?

Reply:
From what I understand, HA core now uses `zh_Hans` and `zh_Hant` for their own loc code - this make sure the translation works for HA installations both pre- and post- `2023.4.x` update

From https://github.com/home-assistant/frontend/issues/16038